core/src/epicli/data/common/ansible/playbooks/roles/repository/files/server/Debian/create-repository.sh

```diff
@@ -1,53 +1,57 @@
 #!/bin/bash -eu
 
-EPI_REPO_SERVER_PATH=$1 # /var/www/html/epirepo is the default
-IS_OFFLINE_MODE=$2
-script_path="$( cd "$(dirname "$0")" ; pwd -P )"
+epi_repo_server_path=$1 # /var/www/html/epirepo is the default
+is_offline_mode=$2 # true/false
+script_path=$(cd "$(dirname "$0")"; pwd -P)
+apt_repo_path="/etc/apt/sources.list"
 
-if $IS_OFFLINE_MODE = true; then
-    # bootstrap apache and dpkg-dev installation in air-gap mode
-    if [[ -f /etc/apt/sources.list ]]; then
-        echo "disabling default repositories..."
-        mv /etc/apt/sources.list /etc/apt/sources.list.bak
-    fi
-    if ! dpkg -l | grep -q libdpkg-perl; then
-        echo libdpkg-perl not found, installing...
-        dpkg -i "${EPI_REPO_SERVER_PATH}"/packages/libdpkg-perl*.deb
-    fi
-    echo "generating repository metadata..."
-    cd "${EPI_REPO_SERVER_PATH}"/packages && /tmp/epi-repository-setup-scripts/dpkg-scanpackages -m . | gzip -9c > Packages.gz && cd "${script_path}"
-    echo "deb [trusted=yes] file:${EPI_REPO_SERVER_PATH}/packages ./" > /etc/apt/sources.list.d/epilocal.list
-    #apt update --assume-no # workaround for botched docker repository https://github.com/docker/for-linux/issues/812
-    echo "updating list of available packages..."
-    apt -y update
-    echo "installing apache..."
-    # force non-interactive mode, ref: https://bugs.launchpad.net/ubuntu/+source/ansible/+bug/1833013
-    DEBIAN_FRONTEND=noninteractive \
-    UCF_FORCE_CONFOLD=1 \
-      apt-get \
-      -o Dpkg::Options::=--force-confold \
-      -o Dpkg::Options::=--force-confdef \
-      -y -q install apache2 dpkg-dev
-    echo "removing temporary repo definition: /etc/apt/sources.list.d/epilocal.list..."
-    rm -f /etc/apt/sources.list.d/epilocal.list
-    #rm -f ${EPI_REPO_SERVER_PATH}/packages/Packages.gz
-    echo "updating list of available packages..."
-    apt -y update
+
+if [[ "$is_offline_mode" == "true" ]]; then
+  # bootstrap apache and dpkg-dev installation in air-gap mode
+  if [[ -f $apt_repo_path ]]; then
+      echo "Disabling default repositories..."
+      mv $apt_repo_path ${apt_repo_path}.bak
+  fi
+
+  [[ $(dpkg -l | grep  "^ii  libdpkg-perl\s") ]] || echo "Package libdpkg-perl not found, installing..."  && \
```

Comment:
```suggestion
  [[ $(dpkg -l | grep -q  "^ii  libdpkg-perl\s") ]] || echo "Package libdpkg-perl not found, installing..."  && \
```

Reply:
I've checked it. 
In case of package is already installed:
```
~ $ if [[ $(dpkg -l | grep "^ii  libdpkg-perl\s") ]]; then echo "Package installed!"; else echo "Package libdpkg-perl not found, installing..."; fi
Package installed!
~ $ if [[ $(dpkg -l | grep -q "^ii  libdpkg-perl\s") ]]; then echo "Package installed!"; else echo "Package libdpkg-perl not found, installing..."; fi
Package libdpkg-perl not found, installing...
```
we don't need --quite option because [[ extension ]] will return 0 or 1.
with --quiet grep will always return 0:
`Exit immediately with zero status if any match is found, even if an error was detected. `
but, I see a different issue right here, If we want to use && as and after conditional checking we have to use parenthesis.